pkg/controller/admissionchecks/multikueue/batchjob_adapter.go

```diff
@@ -97,10 +90,35 @@ func (b *batchJobAdapter) DeleteRemoteObject(ctx context.Context, remoteClient c
 	return client.IgnoreNotFound(remoteClient.Delete(ctx, &job))
 }
 
-func (b *batchJobAdapter) KeepAdmissionCheckPending() bool {
-	return true
+func (b *batchJobAdapter) IsJobManagedByKueue(ctx context.Context, c client.Client, key types.NamespacedName) (bool, string, error) {
+	job := batchv1.Job{}
+	err := c.Get(ctx, key, &job)
+	if err != nil {
+		return false, "", err
+	}
+	jobControllerName := ptr.Deref(job.Spec.ManagedBy, "")
+	if jobControllerName != ControllerName {
+		return false, fmt.Sprintf("Expecting spec.managedBy to be %q not %q", ControllerName, jobControllerName), nil
+	}
+	return true, "", nil
 }
 
-func (b *batchJobAdapter) IsJobManagedByKueue(_ context.Context, _ client.Client, _ types.NamespacedName) (bool, string, error) {
-	return true, "", nil
+var _ multiKueueWatcher = (*batchJobAdapter)(nil)
+
+func (*batchJobAdapter) GetEmptyList() client.ObjectList {
+	return &batchv1.JobList{}
+}
+
+func (*batchJobAdapter) GetWorkloadKey(o runtime.Object) (types.NamespacedName, error) {
+	job, isJob := o.(*batchv1.Job)
+	if !isJob {
+		return types.NamespacedName{}, errors.New("not a job")
```

Comment:
```suggestion
		return types.NamespacedName{}, errors.New("this %q is not a batch/v1,Kind=Job", klog.KObj(o))
```

This seems to be also not used anywhere.